nix/kexec-installer/kexec-run.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/sh
+
+set -ex
+# provided by nix
+init=@init@
+kernelParams=@kernelParams@
+
+SCRIPT_DIR=$(dirname "$(readlink -f "$0")")
+INITRD_TMP=$(TMPDIR=$SCRIPT_DIR mktemp -d)
+
+cd "$INITRD_TMP"
+cleanup() {
+  rm -rf "$INITRD_TMP"
+}
+trap cleanup EXIT
+mkdir -p ssh
+
+extractPubKeys() {
+  home="$1"
+  for file in .ssh/authorized_keys .ssh/authorized_keys2; do
+    key="$home/$file"
+    if test -e "$key"; then
+      # workaround for debian shenanigans
+      grep -o '\(ssh-[^ ]* .*\)' "$key" >> ssh/authorized_keys || true
+    fi
+  done
+}
+extractPubKeys /root
+
+if test -n "${SUDO_USER-}"; then
+  sudo_home=$(sh -c "echo ~$SUDO_USER")
+  extractPubKeys "$sudo_home"
+fi
+
+# Typically for NixOS
+if test -e /etc/ssh/authorized_keys.d/root; then
```

Comment:
I also noticed that on some machines it would still use dash to run this script.
This version is now hopefully more posix compliant.